File: python/minmax.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  minmax.py
#  

import random

def losuj(ileliczb, maksliczb):
    liczby = []  # zbiór pusty

    ile = 0
    # for i in range(ileliczb):
    while ile < ileliczb:
        liczba = random.randint(0, maksliczb)
        if liczby.count(liczba) == 0:
            liczby.append(liczba)
            ile += 1
    #  print(lista)
    return liczby
    
def minimum(lista):
    #  wyszukiwanie minimum
    min = lista[0]
    #  for i, el in enumerate(lista):
    for el in lista:
        if el < min :
            min = el
    return min
    
def maksimum(lista):
    max = lista[0]
    for i, el in enumerate(lista):
        if el > max:
            max = el
    return max
        
def main(args):
    lista = losuj(10, 50)
    assert minimum ([7,5,2,1,2]) == 1
    assert maksimum ([7,5,2,1,2]) == 7
    print("Min: ", minimum(lista))
    print("Max: ", maksimum(lista))
    print(lista)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/minmax.cpp
/*
 * minmax.cpp
 */


#include <iostream>
#include <ctime>
#include <cstdlib>

using namespace std;

int main()
{
    srand( time( NULL ) );
    int tab[ 6 ];
    int x;
    int min, max;
 
    cout << "Wylosowane liczby: " << endl << endl;
    for( int i = 0; i < 6; i++ )
    {
        do
        {
            x =( rand() % 49 ) + 1;
        }
        while(( tab[ 0 ] == x )||( tab[ 1 ] == x )||( tab[ 2 ] == x )||( tab[ 3 ] == x )||( tab[ 4 ] == x )||( tab[ 5 ] == x ) );
 
        tab[ i ] = x;
 
        cout << tab[ i ] << " ";
    }
 
     min = tab[0];
 
    for(int i=0; i<6; i++)
    {
        if(min>tab[i])
        {
            min = tab[i];
        }
    }
 
    max = tab[0];
 
    for(int i=0; i<6; i++)
    {
        if(max<tab[i])
        {
            max = tab[i];
        }
    }
    cout<<endl;
    cout<<"najmniejsza liczba to: "<<min<<endl;
    cout<<"najwieksza liczba to: "<<max<<endl;
    return 0;
}
 
